iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -811,6 +811,25 @@ bool DispatchWorkgroupsOp::canClosureContainOp(Operation *op) {
   return canDispatchRegionContainOp(op);
 }
 
+bool DispatchWorkgroupsOp::isOutputReadWithinRegion(unsigned resultIndex) {
+  BlockArgument arg =
+      body().front().getArgument(operands().size() + resultIndex);
+  // If argument is of `writeonly` access, then it is not read by constructions.
+  if (arg.getType().cast<DispatchTensorType>().getAccess() ==
```

Comment:
style: multi-line ifs should have {}
https://google.github.io/styleguide/cppguide.html#Conditionals